images/router/haproxy/conf/haproxy-config.template

```diff
@@ -612,6 +661,23 @@ backend {{ genBackendNamePrefix $cfg.TLSTermination }}:{{ $cfgIdx }}
           {{- end }}{{/* hsts header */}}
         {{- end }}{{/* is "edge" or "reencrypt" */}}
 
+        {{- if matchValues (print $cfg.TLSTermination) "edge" "reencrypt" }}
```

Comment:
Why do you only need to check for `TLSTermination` type here? Why not in the other places?

Reply:
Also, I understand that the cfg.HTTP*Headers are from Route.spec.  I think you may need to find some way to merge the header manipulation instructions given by IngressSpec with those given in the Route spec.  What if the cluster admin adds a header and the Route.spec deletes it?

Reply:
I think you need to do something like what was done for hsts, see https://github.com/openshift/router/blob/master/images/router/haproxy/conf/haproxy-config.template#L610-L611

Reply:
we have already discussed this that route owner will be able to over-ride the header if same header was redefined.

Reply:
Well, what if this template adds a header, and Route.spec deletes it?  Such as those in https://github.com/openshift/router/pull/438/files#diff-21bd64a4ad8a7927e9d59e99e09628efc884fe6d6291e50eb33c5deb3d2a79f0R354-R364
Shouldn't those be off-limits?

Reply:
Route owner can modify the header at the app level only as he can't modify the front end sections in the haproxy.template. In short the scope for the route owner to set this is at backend level i.e for a specific route which points to a specific backend.

Reply:
Also another option is we can add this code just before set-header stanzas to avoid over-ride of headers set by template.

Reply:
Whatever you do here should be reflected in the enhancement proposal.  I don't think this last option is any better, because the user may still try to set a header that they are not permitted to modify/delete.  IMO, there should be a validation response that tells the user when they cannot modify/delete a specific header, as there is already for HSTS, but the whole list should be there.  If not, then explain why not in the enhancement proposal.

If we don't care about the user experience in trying to modify/delete headers that they shouldn't be permitted to, then put that in the enhancement proposal.

Reply:
added in EP